helper/pool/pool.go

```diff
@@ -427,6 +429,13 @@ func (p *ConnPool) RPC(region string, addr net.Addr, version int, method string,
 		sc.Close()
 		p.releaseConn(conn)
 
+		// If we read EOF, the session is toast. Clear it and open a
+		// new session next time
+		// See https://github.com/hashicorp/consul/blob/v1.6.3/agent/pool/pool.go#L471-L477
+		if lib.IsErrEOF(err) {
+			p.clearConn(conn)
```

Comment:
In `getClient` above on L406 we call release and clear in the opposite order. Looking at the code I don't think it matters and calling `releaseConn` doesn't even seem necessary if we're calling `clearConn`.

It might be nice to make the 2 clear+releaseConn cases the same, but it's not a blocker. I don't see any reason why either would be unsafe.

Reply:
Done - also your proposed order matches Consul's code too.